macros/src/check.rs

```diff
@@ -106,10 +106,16 @@ pub fn app(app: &App) -> parse::Result<()> {
     }
 
     // Check that free interrupts are not being used
-    for int in app.interrupts.keys() {
-        if app.free_interrupts.contains_key(int) {
+    for (name, interrupt) in &app.interrupts {
+        let name = if let Some(ref binds) = interrupt.args.binds {
+            binds
+        } else {
+            name
+        };
```

Comment:
You wrote that (or on the form `.as_ref().unwrap_or()`) 5 times. Maybe it would be better to abstract over this? Maybe there is somewhere that such a pattern is forgotten?

Reply:
Done. I turned the `binds` into a private field and made an accessor method that requires passing the exception / interrupt name.